core/domain/feature_gating_services.py

```diff
@@ -0,0 +1,191 @@
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The service for gating features.
+
+This service provides two interfaces to access the feature flag values for
+clients and servers respectively.
+
+For clients, please use 'evaluate_all_feature_flag_value_for_client' from
+request handlers with client context.
+
+For servers, please directly call 'evaluate_feature_flag_value_for_server' with
+the name of the feature.
+
+For more details of the usage of these two method, please refer their docstrings
+in this file.
+"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core import platform_feature_list
+from core.domain import platform_parameter_domain as param_domain
+from core.domain import platform_parameter_registry as registry
+
+
+ALL_FEATURES_LIST = (
+    platform_feature_list.DEV_FEATURES_LIST +
+    platform_feature_list.TEST_FEATURES_LIST +
+    platform_feature_list.PROD_FEATURES_LIST
+)
+
+ALL_FEATURES_NAMES_SET = set(ALL_FEATURES_LIST)
+
+
+def create_evaluation_context_for_client(client_context_dict):
+    """Returns context instance for evaluation, using the information
+    provided by clients.
+
+    Args:
+        client_context_dict: dict. The client side context.
+
+    Returns:
+        EvaluationContext. The context for evaluation.
+    """
+    return param_domain.EvaluationContext.from_dict(
+        client_context_dict,
+        {
+            'server_mode': _get_server_mode()
+        }
+    )
+
+
+def get_all_feature_flag_dicts():
+    """Returns dict representations of all feature flags.
+
+    Returns:
+        dict. The keys are the feature names and the values are the dict
+        mappings of all fields of the feature flags.
+    """
+    return [
+        registry.Registry.get_platform_parameter(name).to_dict()
+        for name in ALL_FEATURES_LIST
+    ]
+
+
+def evaluate_all_feature_flag_value_for_client(context):
+    """Evaluates and returns the values for all feature flags.
+
+    Args:
+        context: EvaluationContext. The context used for evaluation.
+
+    Returns:
+        dict. The keys are the feature names and the values are boolean
+        results of corresponding flags.
+    """
+    return _evaluate_feature_flag_values_for_context(
+        ALL_FEATURES_LIST, context)
+
+
+def evaluate_feature_flag_value_for_server(feature_name):
+    """Evaluates and returns the values of the feature flag, using context
+    from the server only.
+
+    Args:
+        feature_name: str. The name of the feature flag that needs to
+            be evaluated.
+
+    Returns:
+        bool. The value of the feature flag, True if it's enabled.
```

Comment:
Considering the docstring of the `return` of this function the naming of this function looks complicated. Will it be fine to call it `is_server_feature_enabled(feature_name)` or `is_feature_enabled_for_server(feature_name`?